src/stream/ResponseCollector.jl

```diff
@@ -1,9 +1,90 @@
 # This file is a part of GraknClient.  License is MIT: https://github.com/Humans-of-Julia/GraknClient.jl/blob/main/LICENSE
 
-mutable struct ResponseCollector
-    collectors::Dict{UUID,Queue}
+struct ResponseCollector
+    collectors::Dict{Bytes,Channel{Transaction_Res_All}}
+    transact_result_channel::Channel{Proto.Transaction_Server}
+    access_lock::ReentrantLock
 end
 
+
+function ResponseCollector(transact_result_channel::Channel{Proto.Transaction_Server})
+    collectors = Dict{Bytes,Channel{Transaction_Res_All}}()
+    access_lock = ReentrantLock()
+    resp_col = ResponseCollector(collectors, transact_result_channel, access_lock)
+    res_task = @async response_worker(resp_col)
+    return resp_col
+end
+
+"""
+newId_result_channel(resp_collector::ResponsCollector, request::T) where {T<:Proto.ProtoType}
+    Function is meant to give back the result_channel in which the results for one request
+    will be collected.
+    Attention! Don't put a new Id manually on the ResponsCollector. It wouldn't be thread safe
+"""
+function newId_result_channel(resp_collector::ResponseCollector, req_id::Bytes)
+    res_channel = Channel{Transaction_Res_All}(10)
+    try
+        lock(resp_collector.access_lock)
+        resp_collector.collectors[req_id] = res_channel
+    catch
+    finally
+        unlock(resp_collector.access_lock)
+    end
+    return res_channel
+end
+
+"""
+function remove_Id(resp_collector::ResponsCollector, id::Bytes)
+    The function will close the collecting result channel and remove
+    this from the the resonse collector.
+    Attention! Don't remove a result_channel manually from the Dictionary.
+    This will not be thread safe.
+"""
+function remove_Id(resp_collector::ResponseCollector, id::Bytes)
```

Comment:
This looks like a good case to extend `Base.delete!` function

Reply:
I will integerste this into the next iteration.

Reply:
Done.